src/api/event-registration/service.ts

```diff
@@ -40,7 +38,7 @@ export async function createUniqueRegistrationNumberFromBundle(
   return {
     trackingId,
     registrationNumber: generateRegistrationNumber(trackingId),
-    compositionId: getCompositionId(bundle),
+    compositionId: getCompositionId(bundle)!,
```

Comment:
Can you make `getCompositionId` so it throws if there's no id so you don't have to mark it with `!` here? It's named **get**CompositionId after all.

Reply:
👍 True, we could even just add
```
export function getCompositionId(resBody: fhir.Bundle) {
  return resBody.entry
    ?.map((e) => e.resource)
    .find((res) => res?.resourceType === 'Composition')?.id
}
```

`?` --> `!` here and it would throw if there is not an id

Reply:
I don't think composition can really have an undefined id in our case. 

`get` prefix for a function name should always indicate that there is a return value or otherwise the function throws. So you can make the function throw if there's no composition or its id found

Reply:
This utility could be imported from @opencrvs/commons 🤔 Wonder what effort would it be to publish it for 1.7

Reply:
Yea, sometime in the future. For now it would increase the scope a bit too much